src/dialect/mod.rs

```diff
@@ -38,4 +39,8 @@ pub trait Dialect: Debug {
     fn is_identifier_start(&self, ch: char) -> bool;
     /// Determine if a character is a valid unquoted identifier character
     fn is_identifier_part(&self, ch: char) -> bool;
+
+    fn alllow_single_table_in_parenthesis(&self) -> bool {
+        false
+    }
```

Comment:
Please note https://github.com/ballista-compute/sqlparser-rs/issues/241#issuecomment-664377731

Reply:
Sorry :| -  I don't understand what you mean by that ..  
(to note #241 in code comment ? or something in the code is not aligned with what you commented in [#241 (comment) ](https://github.com/ballista-compute/sqlparser-rs/issues/241#issuecomment-664377731)  )
Can you please elborate ?

Reply:
In that comment I tried to argue for using `if self.dialect <is snowflake>` checks rather than `self.dialect.alllow_single_table_in_parenthesis()`, at least until we better understand when to use which approach.